src/aks-preview/azext_aks_preview/agent/agent.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+import os
+import socket
+import sys
+import uuid
+from pathlib import Path
+
+from azext_aks_preview._consts import (
+    CONST_AGENT_CONFIG_PATH_DIR_ENV_KEY,
+    CONST_AGENT_NAME,
+    CONST_AGENT_NAME_ENV_KEY,
+)
+from azure.cli.core.api import get_config_dir
+from azure.cli.core.commands.client_factory import get_subscription_id
+from knack.util import CLIError
+
+from .prompt import AKS_CONTEXT_PROMPT
+from .telemetry import CLITelemetryClient
+
+
+# NOTE(mainred): holmes leverage the log handler RichHandler to provide colorful, readable and well-formatted logs
+# making the interactive mode more user-friendly.
+# And we removed exising log handlers to avoid duplicate logs.
+# Also make the console log consistent, we remove the telemetry and data logger to skip redundant logs.
+def init_log():
+    # NOTE(mainred): we need to disable INFO logs from LiteLLM before LiteLLM library is loaded, to avoid logging the
+    # debug logs from heading of LiteLLM.
+    logging.getLogger("LiteLLM").setLevel(logging.WARNING)
+    logging.getLogger("telemetry.main").setLevel(logging.WARNING)
+    logging.getLogger("telemetry.process").setLevel(logging.WARNING)
+    logging.getLogger("telemetry.save").setLevel(logging.WARNING)
+    logging.getLogger("telemetry.client").setLevel(logging.WARNING)
+    logging.getLogger("az_command_data_logger").setLevel(logging.WARNING)
+
+    from holmes.utils.console.logging import init_logging
+
+    # TODO: make log verbose configurable, currently disabled by [].
+    return init_logging([])
+
+
+# pylint: disable=too-many-locals
+def aks_agent(
+    cmd,
+    resource_group_name,
+    name,
+    prompt,
+    model,
+    api_key,
+    max_steps,
+    config_file,
+    no_interactive,
+    no_echo_request,
+    show_tool_output,
+    refresh_toolsets,
+):
+    """
+    Interact with the AKS agent using a prompt or piped input.
+
+    :param prompt: The prompt to send to the agent.
+    :type prompt: str
+    :param model: The model to use for the LLM.
+    :type model: str
+    :param max_steps: Maximum number of steps to take.
+    :type max_steps: int
+    :param config_file: Path to the config file.
+    :type config_file: str
+    :param no_interactive: Disable interactive mode.
+    :type no_interactive: bool
+    :param no_echo_request: Disable echoing back the question provided to AKS Agent in the output.
+    :type no_echo_request: bool
+    :param show_tool_output: Whether to show tool output.
+    :type show_tool_output: bool
+    :param refresh_toolsets: Refresh the toolsets status.
+    :type refresh_toolsets: bool
+    """
+    with CLITelemetryClient():
```

Comment:
AKS wants to leverage its own monitoring to collect metrics more than cli startup as well as tool calling and token usages etc. in the future.